tests/unit/test_json_schema_generator.py

```diff
@@ -0,0 +1,875 @@
+                            "required": ["property_name"],
+                        },
+                    }
+                ]
+            ),
+        ),
+        (
+            {"property_name": ["enum1"]},
+            {"enum1": ["rev_dep_property1", "rev_dep_property2"]},
+            JSONSchema(
+                all_of=[
+                    {
+                        "if": {
+                            "properties": {"rev_dep_property1": {"enum": ["enum1"]}}
+                        },
+                        "then": {
+                            "properties": {"property_name": {"not": {"type": "null"}}},
+                            "required": ["property_name"],
+                        },
+                    },
+                    {
+                        "if": {
+                            "properties": {"rev_dep_property2": {"enum": ["enum1"]}}
+                        },
+                        "then": {
+                            "properties": {"property_name": {"not": {"type": "null"}}},
+                            "required": ["property_name"],
+                        },
+                    },
+                ]
+            ),
+        ),
+        (
+            {"property_name": ["enum1", "enum2"]},
+            {"enum1": ["rev_dep_property1"], "enum2": ["rev_dep_property2"]},
+            JSONSchema(
+                all_of=[
+                    {
+                        "if": {
+                            "properties": {"rev_dep_property1": {"enum": ["enum1"]}}
+                        },
+                        "then": {
+                            "properties": {"property_name": {"not": {"type": "null"}}},
+                            "required": ["property_name"],
+                        },
+                    },
+                    {
+                        "if": {
+                            "properties": {"rev_dep_property2": {"enum": ["enum2"]}}
+                        },
+                        "then": {
+                            "properties": {"property_name": {"not": {"type": "null"}}},
+                            "required": ["property_name"],
+                        },
+                    },
+                ]
+            ),
+        ),
+    ],
+    ids=["one rev dep, one enum", "two rev deps, one enum", "two rev deps, two enums"],
+)
+def test_set_conditional_dependencies(
+    reverse_dependencies: dict[str, list[str]],
+    range_domain_map: dict[str, list[str]],
+    expected_schema: dict[str, Any],
+) -> None:
+    """Tests for _set_conditional_dependencies"""
+    json_schema = JSONSchema()
+    _set_conditional_dependencies(
+        json_schema=json_schema,
+        conditional_property="property_name",
+        property_display_name="property_name",
+        reverse_dependencies=reverse_dependencies,
+        range_domain_map=range_domain_map,
+    )
+    assert json_schema == expected_schema
```

Comment:
This seems like an incorrect assertion. Types wouldn't match since a JSONSchema class is not  a `dict[str, Any]`